packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -53,23 +54,32 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
 
     deinitialize(): Promise<void> {
         this._isCredit = undefined;
+        this._isVenmo = undefined;
 
         return Promise.resolve();
     }
 
     private _handleClickButtonProvider({ fundingSource }: ClickDataOptions): void {
         this._isCredit = fundingSource === 'credit' || fundingSource === 'paylater';
+        this._isVenmo = fundingSource === 'venmo';
     }
 
     private _tokenizePayment({ orderID }: ApproveDataOptions) {
         if (!orderID) {
             throw new MissingDataError(MissingDataErrorType.MissingPayment);
         }
+        let provider;
+
+        if (this._isVenmo) {
+            provider = 'paypalcommercevenmo';
+        } else {
+            provider = this._isCredit ? 'paypalcommercecredit' : 'paypalcommerce';
+        }
```

Comment:
It might be better to make it like this:

```suggestion
        let provider;

        if (this._isVenmo) {
            provider = 'paypalcommercevenmo';
        } else if (this._isCredit) {
            provider = 'paypalcommercecredit';
        } else {
            provider = 'paypalcommerce';
        }
```